source/common/tcp_proxy/upstream.cc

```diff
@@ -526,7 +523,24 @@ void CombinedUpstream::onUpstreamTrailers(Http::ResponseTrailerMapPtr&& trailers
   responseDecoder().decodeTrailers(std::move(trailers));
 }
 
-Http::RequestHeaderMap* CombinedUpstream::downstreamHeaders() { return downstream_headers_.get(); }
+Http::RequestHeaderMap* CombinedUpstream::downstreamHeaders() {
+  if (!downstream_headers_) {
```

Comment:
when will this `!downstream_headers` happen considering there is one definition in the constructor?